phpstan.neon

```diff
@@ -275,7 +275,7 @@ parameters:
         -
             message: '#Node "switch" is fobidden to use#'
             paths:
-                 - packages/easy-hydrator/src/ObjectCreator.php
+                 - packages/easy-hydrator/src/HydratedObjectFactory.php
```

Comment:
This is the last challange here. Could you remove this and refactor Switch to collector?
It would allow better refactoring in the future.

Reply:
Yeah, i will try to decouple and refactor to collection of "retypers", quick idea:
```php
interface Retyper {
    public function isTypeSupported(string $type): bool;

    public function retype(mixed $data): mixed;
}
```

```php
foreach ($retypers as $retyper) {
    if ($retyper->isTypeSupported($type)) {
        return $retyper->retype($data);
    }
}
```

Reply:
Purfect :+1:

Reply:
@TomasVotruba pls check https://github.com/symplify/symplify/pull/2195/commits/1d9860f7c728aa61927b2f21009f5a04c3ccbefb and https://github.com/symplify/symplify/pull/2195/commits/026231e53417f12877af2fb91c89a28f493b4ddd just to confirm this is the right way to go. I will get to full implementation later today, so just confirmation now is enough :-)